rust/operator-binary/src/main.rs

```diff
@@ -127,15 +123,27 @@ async fn main() -> anyhow::Result<()> {
 }
 
 fn references_authentication_class(
-    authentication_config: &AirflowAuthentication,
-    authentication_class: &AuthenticationClass,
+    airflow: &DeserializeGuard<AirflowCluster>,
+    authentication_class: &DeserializeGuard<AuthenticationClass>,
 ) -> bool {
-    assert!(authentication_class.metadata.name.is_some());
+    airflow
+        .0
+        .as_ref()
+        .expect("AirflowCluster object is invalid");
+    authentication_class
+        .0
+        .as_ref()
+        .expect("AuthenticationClass object is invalid");
```

Comment:
We shouldn't `expect` here, just return `false` (and maybe log an error).

Reply:
Actually, we don't care about the authnclass being valid here since `DeserializeGuard` still provides object metadata.

Invalid airflows would still trigger the main `airflow_controller` so we should be fine just skipping that.

Reply:
Even better! [Thanks](https://github.com/stackabletech/airflow-operator/pull/520/commits/c342854bc41688ec98faf17af058165350c75a32)

Reply:
Oh, we still want to filter by the authnclass. Gimme a moment.

Reply:
```patch
diff --git a/rust/operator-binary/src/main.rs b/rust/operator-binary/src/main.rs
index d3b85c1..63b9f43 100644
--- a/rust/operator-binary/src/main.rs
+++ b/rust/operator-binary/src/main.rs
@@ -17,7 +17,10 @@ use stackable_operator::{
     k8s_openapi::api::{apps::v1::StatefulSet, core::v1::Service},
     kube::{
         core::DeserializeGuard,
-        runtime::{reflector::ObjectRef, watcher, Controller},
+        runtime::{
+            reflector::{Lookup, ObjectRef},
+            watcher, Controller,
+        },
         ResourceExt,
     },
     logging::controller::report_controller_reconciled,
@@ -126,24 +129,16 @@ fn references_authentication_class(
     airflow: &DeserializeGuard<AirflowCluster>,
     authentication_class: &DeserializeGuard<AuthenticationClass>,
 ) -> bool {
+    let Ok(airflow) = &airflow.0 else {
+        return false;
+    };
+    let Some(authn_class_name) = authentication_class.name() else {
+        return false;
+    };
     airflow
-        .0
-        .as_ref()
-        .expect("AirflowCluster object is invalid");
-    authentication_class
-        .0
-        .as_ref()
-        .expect("AuthenticationClass object is invalid");
-
-    if let Ok(airflow_cluster) = airflow.0.as_ref() {
-        let authentication_config = airflow_cluster.spec.clone().cluster_config.authentication;
-        authentication_config
-            .authentication_class_names()
-            .into_iter()
-            .filter(|c| *c == authentication_class.name_any())
-            .count()
-            > 0
-    } else {
-        false
-    }
+        .spec
+        .cluster_config
+        .authentication
+        .authentication_class_names()
+        .contains(&&*authn_class_name)
 }
```

Reply:
:+1: